File: backoffice/src/product/domain/events/ProductDeleted.py
"""
 *
 * Libraries 
 *
"""

from ..valueObjects.ProductId import ProductId
from src.shared.domain        import DomainEvent

"""
 *
 * Class
 *
"""

class ProductDeleted( DomainEvent ):

    """
     *
     * Attributes 
     *
    """

    __id : ProductId

    """
     *
     * Methods 
     *
    """

    def __init__( 
        self, 
        id        : ProductId   = None,
        eventId   : str         = None,
        timestamp : int         = None
    ) -> None:
        super().__init__( eventId, timestamp )
        self.__id   = id
    
    def getId( self ) -> ProductId:
        return self.__id
    
    def getEventName( self ) -> str:
        return 'product.deleted'
    
    def toPrimitives( self ) -> dict[str, str | int]:
        # Variables
        data : dict[str, str | int]
        # Code
        data = {
            'id' : self.__id.getValue()
        }
        return data

    def fromPrimitives( self, 
        eventId   : str, 
        timestamp : int, 
        data      : dict[str, str | int]
    ) -> object:
        # Variables
        domainEvent : object
        # Code
        domainEvent = ProductDeleted(
            id        = ProductId( data['id'] ),
            eventId   = eventId,
            timestamp = timestamp
        )
        return domainEvent

File: preparation/src/product/domain/events/ProductDeleted.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class ProductDeleted : DomainEvent {

        /* 
         * 
         * Attributes
         *
        */

        private ProductId id;

        /* 
         * 
         * Methods
         *
        */

        public ProductDeleted() {
        }

        public ProductDeleted( ProductId id ) {
            this.id = id;
        }

        public ProductDeleted(
            ProductId id,
            string    eventId,
            int       timestamp
        ) : base( eventId, timestamp ) {
            this.id = id;
        }

        public ProductId getId() {
            return id;
        }

        public override DomainEvent fromPrimitives(
            string                     eventId, 
            int                        timestamp, 
            Dictionary<string, object> data
        ) {
            // Variables
            ProductDeleted domainEvent;
            // Code
            domainEvent = new ProductDeleted(
                new ProductId( data["id"].ToString() ),
                eventId,
                timestamp
            );
            return domainEvent;
        }

        public override string getEventName() {
            return "product.deleted";
        }

        public override Dictionary<string, object> toPrimitives() {
            // Variables
            Dictionary<string, object> data;
            // Code
            data = new Dictionary<string, object>();
            data.Add( "id", id.getValue() );
            return data;
        }

    }

}